### PR TITLE
feat: comment in PR when branch automerge fails

### LINKED ISSUE #1934: Add comment to PR when branch automerge fails
It would be useful to tell users why the PR was raised. Can’t be in PR body itself as it would get overridden the next run, so we should use the ensure issue comment capability to add it after the PR is created, like we do with assignees, reviewers and labels.

It should have the heading "Branch automerge failure" and then the content can say "This PR was configured for branch automerge, however this is not possible so it has been raised as a PR instead." Ideally we could pass an explanation through to add to it, e.g. if it's because it timed out from pending then the explanation is "Status checks remained in pending state for too long". If there are status check failures then the explanation is "Branch has one or more failed status checks" and ideally a stringified copy of the status check response.

### PR DESCRIPTION
**Pull request by dragomirt**
This PR adds the feature of commenting on a failed automerge. It's done by adding a conditional in `lib/workers/branch/automerge.js` which, in case of receiving `failure` or `error` from the `getBranchStatus` function, returns the "branch status error" value. Another modification is in `lib/workers/branch/index.js`, which is an adition to the failure response of the `tryBranchAutomerge` function. The added functionality is the ability to add a comment to the PR which had a failure automerging. In case of receiving the aforementioned "branch status error" value, to the comment is appended a note which emphasize the fact that there're multiple failed status checks.

Closes #1934

### CODE REVIEW

File: lib/workers/branch/index.js [hunk 1]
```diff
@@ -177,10 +177,24 @@ async function processBranch(branchConfig, packageFiles) {
...
     } else if (
       mergeStatus === 'automerge aborted - PR exists' ||
-      mergeStatus === 'failed'
+      mergeStatus === 'failed' ||
+      mergeStatus === 'branch status error'
     ) {
       logger.info({ mergeStatus }, 'Branch automerge not possible');
       config.forcePr = true;
+      // istanbul ignore next
+      (async () => {
```
Comment (rarkins) [inline lib/workers/branch/index.js]: can we avoid this by having the full function be async instead?

**Comment by dragomirt** [comment]
It works fine on GitHub, but as it seems, there is no `ensureComment` function for GitLab (https://github.com/renovateapp/renovate/blob/9fbaf9a3ca313429c9fc0d29ae892306d89559ff/lib/platform/gitlab/index.js#L384). Working on that.

**Comment by rarkins** [comment]
If you add ensureComment for GitLab, please make it a separate PR that we merge first. It would be useful

File: lib/workers/branch/index.js [hunk 2]
```diff
@@ -172,15 +172,28 @@ async function processBranch(branchConfig, packageFiles) {
...
       return 'automerged';
     } else if (
       mergeStatus === 'automerge aborted - PR exists' ||
-      mergeStatus === 'failed'
+      mergeStatus === 'failed' ||
+      mergeStatus === 'branch status error'
     ) {
       logger.info({ mergeStatus }, 'Branch automerge not possible');
       config.forcePr = true;
+      const branchPr = await platform.getBranchPr(config.branchName);
```
Comment (rarkins) [inline lib/workers/branch/index.js]: On the first run through, this PR won't exist yet

**Review by rarkins — CHANGES_REQUESTED** [review]
Needs a change in logic/location to happen after initial PR creation.

File: lib/workers/branch/index.js [hunk 3]
```diff
@@ -86,6 +86,16 @@ async function processBranch(branchConfig, packageFiles) {
       const branchPr = await platform.getBranchPr(config.branchName);
       if (branchPr) {
         logger.debug('Found existing branch PR');
+        // istanbul ignore if
```
Comment (rarkins) [inline lib/workers/branch/index.js]: No, this needs to be in PR, not branch. It should be run after the createPr

**Review by rarkins — APPROVED** [review]

**Pull request merged**